receiver/libhoneyreceiver/receiver.go

```diff
@@ -243,16 +252,24 @@ func (r *libhoneyReceiver) handleEvent(resp http.ResponseWriter, req *http.Reque
 
 	numLogs := otlpLogs.LogRecordCount()
 	if numLogs > 0 {
-		ctx = r.obsreport.StartLogsOp(ctx)
-		err = r.nextLogs.ConsumeLogs(ctx, otlpLogs)
-		r.obsreport.EndLogsOp(ctx, "protobuf", numLogs, err)
+		if r.nextLogs != nil {
+			ctx = r.obsreport.StartLogsOp(ctx)
+			err = r.nextLogs.ConsumeLogs(ctx, otlpLogs)
+			r.obsreport.EndLogsOp(ctx, "protobuf", numLogs, err)
+		} else {
+			r.settings.Logger.Debug("Dropping log records - no log consumer configured", zap.Int("dropped_logs", numLogs))
+		}
 	}
 
 	numTraces := otlpTraces.SpanCount()
 	if numTraces > 0 {
-		ctx = r.obsreport.StartTracesOp(ctx)
-		err = r.nextTraces.ConsumeTraces(ctx, otlpTraces)
-		r.obsreport.EndTracesOp(ctx, "protobuf", numTraces, err)
+		if r.nextTraces != nil {
+			ctx = r.obsreport.StartTracesOp(ctx)
+			err = r.nextTraces.ConsumeTraces(ctx, otlpTraces)
+			r.obsreport.EndTracesOp(ctx, "protobuf", numTraces, err)
+		} else {
+			r.settings.Logger.Debug("Dropping trace spans - no trace consumer configured", zap.Int("dropped_spans", numTraces))
```

Comment:
We should probably capture telemetry around these mixed up cases. If the user got here it means their telemetry looked different than they expected or they didn't configure the collector properly. An error is probably appropriate or some logs/metrics recording what happened.